src/browser-frontend/reducers/pages-reducers.js

```diff
@@ -32,7 +32,7 @@ function addPage(state, { payload: { url, index, background } = {} }) {
     mut.updateIn(['ui', 'pages', 'visuals'], m => m.set(pageId, pageUIState));
 
     const pageCount = state.ui.pages.displayOrder.count();
-    const pageIndex = index != null ? index : pageCount;
+    const pageIndex = parentId ? state.ui.pages.displayOrder.findIndex(id => id === parentId) + 1 : pageCount;
```

Comment:
Nit: This doesn't handle the case where `parentId` isn't in `displayOrder`. I bet that can never happen now, but never trust your caller etc.